h1st/model/fuzzy_logic/fuzzy_logic_rules.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from typing import NoReturn
+
+import skfuzzy
+from skfuzzy import control as ctrl
+from skfuzzy.control.term import Term
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class FuzzyLogicRules:
+    def __init__(self) -> None:
+        self.membership = {
+            'triangle': skfuzzy.trimf,
+            'trapezoid': skfuzzy.trapmf,
+            'gaussian': skfuzzy.gaussmf,
+        }
+        self.vars = {}
+        self.rules = {}
+
+    def add_rule(self, name, if_: Term, then_: Term): #  -> NoReturn
+        """
+        Add a fuzzy rule. Place antecedent type variables in 'if' statement
+        and place consequent type variables in 'then' statement.
+
+        .. code-block:: python
+            :caption: example
+            flr = FuzzyLogicRules()
+            flr.add_rule(
+                'rule1',
+                if_=flr.vars['sensor1']['abnormal'],
+                then_=flr.vars['problem1']['yes'])
+        """
+        rule = ctrl.Rule(if_, then_)
+        self.rules[name] = rule
+
+    def remove_rule(self, name: str) -> NoReturn:
+        if name in self.rules:
+            del self.rules[name]
```

Comment:
Recommend using `pop()` (cf. https://stackoverflow.com/questions/15411107/delete-a-dictionary-item-if-the-key-exists)